src/test/java/guitests/PersonEditGuiTest.java

```diff
@@ -42,35 +45,133 @@ public void editPerson() {
 
         //Confirm right values are displayed after grace period is over
-        guiRobot.sleep(5000); //wait for grace period
-        //TODO: link up to grace period
-        //TODO: confirm the person card display behave as expected during grace period
-        assertTrue(descOnFail(alicePersonCard, newAlice), alicePersonCard.isSamePerson(newAlice));
+        mainGui.sleepForGracePeriod();
+        assertEquals(alicePersonCard, newAlice);
 
         //Confirm the underlying person object has the right values
         assertEquals(newAlice.toString(), personListPanel.getSelectedPerson().toString());
 
-        //TODO: confirm again after the next sync
-        //TODO: confirm other cards are unaffected
-        //TODO: confirm the right card is selected after the edit
-        //TODO: confirm status bar is updated correctly
 
+        //Confirm again after the next sync
+        mainGui.sleep(getTestingConfig().getUpdateInterval(), TimeUnit.MILLISECONDS);
+        assertEquals(newAlice.toString(), personListPanel.getSelectedPerson().toString());
+
+        //Confirm other cards are unaffected.
+        personListPanel.clickOnListView();
+        personListPanel.use_LIST_JUMP_TO_INDEX_SHORTCUT(2); //To make sure the card is visible
+        assertEquals(personListPanel.getPersonCardHandle(1), td.benson); //Ensure the card details are correct
+        personListPanel.use_LIST_JUMP_TO_INDEX_SHORTCUT(3);
+        assertEquals(personListPanel.getPersonCardHandle(2), td.charlie);
+        personListPanel.use_LIST_JUMP_TO_INDEX_SHORTCUT(4);
+        assertEquals(personListPanel.getPersonCardHandle(3), td.dan);
+
+        //Confirm status bar is updated correctly
+        assertEquals(statusBar.getText(), "Edit Person [ Alice Brown -> Alicia Brownstone ] completed successfully.");
+    }
+
+
+    @Test
+    public void editPerson_usingContextMenu() {
+        personListPanel.rightClickOnPerson(td.alice);
+        EditPersonDialogHandle editPersonDialog = personListPanel.clickOnContextMenu(
+                PersonListPanelHandle.ContextMenuChoice.EDIT);
+        assertTrue(editPersonDialog.isValidEditDialog());
+    }
+
+
+    @Test
+    public void editPerson_usingEditButton() {
+        personListPanel.clickOnPerson(td.alice);
+        EditPersonDialogHandle editPersonDialog =  personListPanel.clickEdit();
+        assertTrue(editPersonDialog.isValidEditDialog());
+    }
+
+    @Test
+    public void editPerson_dataValidation() {
+
+        personListPanel.clickOnPerson(td.alice);
+        EditPersonDialogHandle editPersonDialog =  personListPanel.clickEdit();
+        editPersonDialog.enterFirstName("Peter");
+        editPersonDialog.enterLastName("");
+        editPersonDialog.clickOk();
+
+        assertTrue(editPersonDialog.isInputValidationErrorDialogShown());
+        editPersonDialog.dissmissErrorMessage("Invalid Fields");
+        assertFalse(editPersonDialog.isInputValidationErrorDialogShown());
+    }
```

Comment:
May be add a TODO for testing edits during grace period? Can be done in a separate PR.